<a href="https://colab.research.google.com/github/anaghasanthoshh/Msc_Project/blob/main/EDA_Notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import matplotlib.pyplot

In [ ]:
data_review=pd.read_json('All_Beauty.jsonl',lines=True)

In [ ]:
data_review.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 701528 entries, 0 to 701527
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype         
---  ------             --------------   -----         
 0   rating             701528 non-null  int64         
 1   title              701528 non-null  object        
 2   text               701528 non-null  object        
 3   images             701528 non-null  object        
 4   asin               701528 non-null  object        
 5   parent_asin        701528 non-null  object        
 6   user_id            701528 non-null  object        
 7   timestamp          701528 non-null  datetime64[ns]
 8   helpful_vote       701528 non-null  int64         
 9   verified_purchase  701528 non-null  bool          
dtypes: bool(1), datetime64[ns](1), int64(2), object(6)
memory usage: 48.8+ MB


In [ ]:
data_review.head()

rating                                      title  \
0       5  Such a lovely scent but not overpowering.   
1       4     Works great but smells a little weird.   
2       5                                       Yes!   
3       1                          Synthetic feeling   
4       5                                         A+   

                                                text images        asin  \
0  This spray is really nice. It smells really go...     []  B00YQ6X8EO   
1  This product does what I need it to do, I just...     []  B081TJ8YS3   
2                          Smells good, feels great!     []  B07PNNCSP9   
3                                     Felt synthetic     []  B09JS339BZ   
4                                            Love it     []  B08BZ63GMJ   

  parent_asin                       user_id               timestamp  \
0  B00YQ6X8EO  AGKHLEW2SOWHNMFQIJGBECAF7INQ 2020-05-05 14:08:48.923   
1  B081TJ8YS3  AGKHLEW2SOWHNMFQIJGBECAF7INQ 2020-05-04 18:10:55.070   
2  B097R46CSY  AE74DYR3QUGVPZJ3P7RFWBGIX7XQ 2020-05-16 21:41:06.052   
3  B09JS339BZ  AFQLNQNQYFWQZPJQZS6V3NZU4QBQ 2022-01-28 18:13:50.220   
4  B08BZ63GMJ  AFQLNQNQYFWQZPJQZS6V3NZU4QBQ 2020-12-30 10:02:43.534   

   helpful_vote  verified_purchase  
0             0               True  
1             1               True  
2             2               True  
3             0               True  
4             0               True

In [ ]:
(data_review.verified_purchase==False).sum()

66559

There are around 66559 reviews that are not from verified purchase.

In [ ]:
pd.average()

In [ ]:
data_product=pd.read_json('meta_All_Beauty.jsonl',lines=True)

In [ ]:
data_product.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112590 entries, 0 to 112589
Data columns (total 14 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   main_category    112590 non-null  object 
 1   title            112590 non-null  object 
 2   average_rating   112590 non-null  float64
 3   rating_number    112590 non-null  int64  
 4   features         112590 non-null  object 
 5   description      112590 non-null  object 
 6   price            17704 non-null   float64
 7   images           112590 non-null  object 
 8   videos           112590 non-null  object 
 9   store            101259 non-null  object 
 10  categories       112590 non-null  object 
 11  details          112590 non-null  object 
 12  parent_asin      112590 non-null  object 
 13  bought_together  0 non-null       float64
dtypes: float64(3), int64(1), object(10)
memory usage: 12.0+ MB


In [ ]:
pd.set_option('display.width', 500)

In [ ]:
data_product.head(15)

main_category                                              title  average_rating  rating_number                                           features                                        description  price                                             images videos                   store categories                                            details parent_asin  bought_together
0     All Beauty  Howard LC0008 Leather Conditioner, 8-Ounce (4-...             4.8             10                                                 []                                                 []    NaN  [{'thumb': 'https://m.media-amazon.com/images/...     []         Howard Products         []  {'Package Dimensions': '7.1 x 5.5 x 3 inches; ...  B01CUPMQZE              NaN
1     All Beauty  Yes to Tomatoes Detoxifying Charcoal Cleanser ...             4.5              3                                                 []                                                 []    NaN  [{'thumb': 'https://m.media-amazon.com/images/...     []                  Yes To         []  {'Item Form': 'Powder', 'Skin Type': 'Acne Pro...  B076WQZGPM              NaN
2     All Beauty   Eye Patch Black Adult with Tie Band (6 Per Pack)             4.4             26                                                 []                                                 []    NaN  [{'thumb': 'https://m.media-amazon.com/images/...     []  Levine Health Products         []         {'Manufacturer': 'Levine Health Products'}  B000B658RI              NaN
3     All Beauty  Tattoo Eyebrow Stickers, Waterproof Eyebrow, 4...             3.1            102                                                 []                                                 []    NaN  [{'thumb': 'https://m.media-amazon.com/images/...     []                Cherioll         []  {'Brand': 'Cherioll', 'Item Form': 'Powder', '...  B088FKY3VD              NaN
4     All Beauty  Precision Plunger Bars for Cartridge Grips – 9...             4.3              7  [Material: 304 Stainless Steel; Brass tip, Len...  [The Precision Plunger Bars are designed to wo...    NaN  [{'thumb': 'https://m.media-amazon.com/images/...     []               Precision         []                            {'UPC': '644287689178'}  B07NGFDN6G              NaN
5     All Beauty  Lurrose 100Pcs Full Cover Fake Toenails Artifi...             3.7             35  [The false toenails are durable with perfect l...  [Description, The false toenails are durable w...   6.99  [{'thumb': 'https://m.media-amazon.com/images/...     []                 Lurrose         []  {'Color': 'As Shown', 'Size': 'Large', 'Materi...  B07G9GWFSM              NaN
6     All Beauty  Stain Bonnet For Baby Bonnet Silk Sleep Cap Fo...             4.1             50                                                 []                                                 []    NaN  [{'thumb': 'https://m.media-amazon.com/images/...     []                Edoneery         []  {'Brand': 'Edoneery', 'Material': 'Silk', 'Num...  B08XZ97HFY              NaN
7     All Beauty  50 Pieces False Eyelash Packaging Box Empty Ey...             3.8             32                                                 []                                                 []    NaN  [{'thumb': 'https://m.media-amazon.com/images/...     []                  Maitys         []  {'Package Dimensions': '14.49 x 11.26 x 2.36 i...  B08DNQTTQK              NaN
8     All Beauty                         Gold extatic Musk EDT 90ml             3.7              2  [Extatic Balmain Gold Musk By Balmain Edt Spra...             [Edt spray 3 oz design house: balmain]  86.95  [{'thumb': 'https://m.media-amazon.com/images/...     []                 Balmain         []  {'Brand': 'Balmain', 'Item Form': 'Spray', 'It...  B01ERJEGS6              NaN
9     All Beauty  4 Pieces Satin Bonnet Adjustable Sleep Cap Dou...             4.3             66                                                 []                                            